### ISSUE TITLE
[Request] ShearsItem should inherit all mechanics of the vanilla shears

### ISSUE THREAD

**Issue by raziel23x** [MinecraftForge/MinecraftForge#7508]
I have noticed when i created a new tool using the ShearsItem to make custom shears i noticed that carved pumkins and the beehives/nest was excluded and i had to do something long the lines of this to give the shears the ability to carve pumkins like the vanilla shears I have yet to work on how to do so with beehives but i would of thought that a tool should be able to do all aspect of the vanilla tool without extra coding like the other tools and weapons which take on everything from there class

```java
@SubscribeEvent
    public void onRightClickBlock(PlayerInteractEvent.RightClickBlock event) {
        World world = event.getWorld();

        if (!world.isRemote) {
            PlayerEntity player = event.getPlayer();
            BlockState state = world.getBlockState(event.getPos());
            Block block = state.getBlock();

            if (event.getItemStack().getItem() == RegistryHandler.FLINT_SHEARS.get()) {
                if (block.getBlock() == Blocks.PUMPKIN) {
                    //LOGGER.info("Clicked Pumpkin with Flint Shears");

                    ItemStack seedItem = new ItemStack (Items.PUMPKIN_SEEDS);
                    seedItem.setCount(4);

                    world.removeBlock(event.getPos(),false);
                    world.setBlockState(event.getPos(), Blocks.CARVED_PUMPKIN.getDefaultState());
                    world.playSound(null, event.getPos(), SoundEvents.BLOCK_PUMPKIN_CARVE, SoundCategory.BLOCKS, 1f, 1f);
                    block.spawnAsEntity(world, event.getPos(), seedItem);

                    if (!player.isCreative()) {
                        if (event.getItemStack().getItem().isDamageable()) {
                            event.getItemStack().getItem().setDamage(event.getItemStack(), event.getItemStack().getItem().getDamage(event.getItemStack()) + 1);
                        }
                    }
                } else {

                    if (block.getBlock() == Blocks.BEE_NEST) {
                        // do something
                    }
                }
            }
        }
    }```

**Comment by diesieben07** [comment]
Forge already has `IForgeShearable`, which those blocks should implement. However the vanilla shears would also need to be patched to actually support `IForgeShearable` on blocks - which it currently does not.

**Comment by percivalalb** [comment]
https://github.com/MinecraftForge/MinecraftForge/commit/b222b23557f2bf42909d4bdd719a014a96100657
This was recently commited so as long as your shears extend `ShearsItem` I believe they will now work on pumpkins and bee hives. So your workaround is no longer needed

**Comment by TheCurle** [comment]
Closed by #7544
